src/pages/[locale]/[...path]/index.astro

```diff
@@ -53,19 +53,30 @@ type Params = {
 
 const { locale, path } = Astro.params as Params;
 const variables = { locale, slug: getPageSlugFromPath(path) };
-const { page } = (await datocmsRequest<PageQuery>({ query, variables })) as {
-  page: NonNullable<PageQuery['page']>; // Only NonNullable when statically generated. Handle as a 404 when this is a server route!
-};
-const breadcrumbs = [...getParentPages(page), page].map((page) =>
+const { page } = await datocmsRequest<PageQuery>({ query, variables });
+
+if (!page) {
```

Comment:
This check ensures this type casting is no longer needed:
```ts
as {
  page: NonNullable<PageQuery['page']>; // Only NonNullable when statically generated. Handle as a 404 when this is a server route!
};
```

But it makes the `404` rewrite and locale setting in that file more complex. Worth it?